### DISCUSSION TITLE
Add postprocessing steps in TransformedTargetForecaster

### DISCUSSION THREAD

**Discussion by eyalshafran** [alan-turing-institute/sktime#1413] [General]
As far as I can tell the TransformedTargetForecaster only accepts a forecaster as the last step. I was wondering if it is possible to incorporate post processing steps in the pipeline. For example, after using the forecaster to predict, one might want to force all the negative values to be 0. Is it possible to do something like that currently? If not, are there any plans to allow for that in the future?

**Comment by mloning** [comment]
Yes, there are currently some discussion around a more advanced pipeline interface, see #1174

**Comment by fkiraly** [comment]
contributions are welcome :-)

**Comment by fkiraly** [comment]
there's also the almost merged network pipeline which should allow you to do this out-of-the-box?
https://github.com/alan-turing-institute/sktime/pull/856

testing this and feedback is also appreciated

FYI @ViktorKaz

**Comment by eyalshafran** [comment]
Thank you for the replies. I will look into the advanced pipeline and network pipeline. 
In the meantime I found a really hacky solution - I can create a custom transformation that just passes the value with transform but does some postprocessing in inverse_transform. I can then use that as the first transformation in the pipeline. It will not change the data during fit_transform but since the inverse_transform is invoked during predict it will change the data then. This is ugly but it works for what I'm trying to do at the moment. 

```
pipe = TransformedTargetForecaster(steps=[
        ("postprocess, PostProcess()),
         ("imputer", Imputer(method="mean")),
        ("detrender", Deseasonalizer()),
       ("forecaster", NaiveForecaster(strategy="drift"))])
```

**Reply by fkiraly** [reply]
@eyalshafran, I've added this in here:
https://github.com/alan-turing-institute/sktime/issues/1174

In the "advanced solutions" where transformers can be chained in a position *after* the forecaster, that should cover your use case precisely, right?
